src/codecovio.jl

```diff
@@ -65,6 +65,94 @@ module Codecov
     on TravisCI or AppVeyor. If running locally, use `submit_local`.
     """
     function submit(fcs::Vector{FileCoverage}; kwargs...)
+        submit_generic(fcs; add_ci_to_kwargs(;kwargs...)...)
+    end
+
+
+    """
+        submit_local(fcs::Vector{FileCoverage}, dir::AbstractString=pwd())
+
+    Take a `Vector` of file coverage results (produced by `process_folder`),
+    and submit them to Codecov.io. Assumes the submission is being made from
+    a local git installation, rooted at `dir`. A repository token should be specified by a
+    `token` keyword argument or the `CODECOV_TOKEN` environment variable.
+    """
+    function submit_local(fcs::Vector{FileCoverage}, dir::AbstractString=pwd(); kwargs...)
+        submit_generic(fcs; add_local_to_kwargs(dir; kwargs...)...)
+    end
+
+    @deprecate submit_token submit_local
+
+
+    """
+        submit_generic(fcs::Vector{FileCoverage})
+
+    Takes a vector of file coverage results (produced by `process_folder`),
+    and submits them to a Codecov.io instance. Keyword arguments are converted
+    into a generic Codecov.io API uri.  It is essential that the keywords and
+    values match the Codecov upload/v2 API specification.
+    The `codecov_url` keyword argument or the CODECOV_URL environment variable
+    can be used to specify the base path of the uri.
+    """
+    function submit_generic(fcs::Vector{FileCoverage}; kwargs...)
```

Comment:
Is there a reason why you are reordering the code? It makes it much harder to review this PR!

Reply:
You are right, I changed the order of some functions. I have been able to reduce that a bit by moving `construct_uri_string()` to the bottom of the file. Still, as I'm basically disentangling two functions, there is inevitably a number of lines being moved.

Reply:
Thanks, that's much better. Of course I realize you can't fully avoid moving some stuff around, but reducing it to a minimum (resp. performing the moving in a separate commit) helps a lot in reviewing. I took the liberty to move some more stuff down, and rebase it to resolve conflicts, squashing it into one commit. Let's see if the tests pass.